src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,16 +1,88 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
+@RequiredArgsConstructor
 public class BookController {
 
+    private final String ID_NOT_FOUND = "Book not found";
+    private final BookRepository repository;
+
+    /**
+     * this method create a greeting with a name or for default is world
+     * @param name : Name of the sender (String)
+     *
+    */
     @GetMapping("/greeting")
     public String greeting(@RequestParam(name="name",required = false,defaultValue = "World") String name, Model model){
         model.addAttribute("name",name);
         return "greeting";
     }
+
+
+    /**
+     * this method find all books
+     */
+    @GetMapping
+    public Iterable findAll() {
+        return repository.findAll();
+    }
+
+
+    /**
+     * this method save the book
+     * @param book = object of type(Book)
+     */
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        return repository.save(book);
+    }
+
+
+    /**
+     * this method delete the book by id
+     * @param id = id for delete the book
+     */
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        repository.findById(id)
+                .orElseThrow(() -> new BookNotFoundException(ID_NOT_FOUND) );
+        repository.deleteById(id);
+    }
+
+
+    /**
+     * this method updated the book by id
+     * @param bookNew = object of type Book
+     * @param id = id for update the book (must be equals to Book.id)
+     */
+    @PutMapping("/{id}")
+    public Book updateBook(@RequestBody Book bookNew, @PathVariable Long id) {
+        if(bookNew.getId() != id) {
```

Comment:
Since you're using Long (with uppercase, the class, not the primitive type) the best practice is to use the .equals() method rather than the comparison with == or != 🙂 
The IDE might highlight it as a code smell, too 
This is just a suggestion though 😅